### PR TITLE
feat(availableBalance): added available balance check on helioPay FEE…

### PR DESCRIPTION
**Pull request by TigranHelio**
…-723

### CODE REVIEW

File: packages/react/src/components/baseCheckout/index.tsx [hunk 1]
```diff
@@ -213,9 +237,29 @@ const BaseCheckout = ({
                       setFieldValue={setFieldValue}
                     />
 
-                    <Button type="submit" disabled={payButtonDisable}>
+                    <ButtonWithTooltip
+                      type="submit"
+                      disabled={
+                        payButtonDisable ||
+                        !getIsBalanceEnough(
```
Comment (SurenAvagyanG) [inline packages/react/src/components/baseCheckout/index.tsx:244]: boolean function naming is better with is or has prefixes. let's have isBalanceEnough here

File: packages/react/src/components/baseCheckout/index.tsx [hunk 2]
```diff
@@ -103,6 +103,30 @@ const BaseCheckout = ({
...
   );
 
+  const getIsBalanceEnough = useCallback(
+    (
+      customPrice?: number,
+      quantity?: number,
+      currency?: string,
+      interval?: number
+    ) =>
+      AvailableBalanceService.getIsBalanceEnough({
```
Comment (SurenAvagyanG) [inline packages/react/src/components/baseCheckout/index.tsx:113]: boolean function naming is better with is or has prefixes. let's have isBalanceEnough here

File: packages/react/src/domain/services/AvailableBalanceService.ts
Comment (SurenAvagyanG) [inline packages/react/src/domain/services/AvailableBalanceService.ts:1]: Move this service with it's models to HelioSDK
under HelioSDK class you will have 
this.availablebalanceService = new AvailableBalanceService(...)

**Review by SurenAvagyanG — CHANGES_REQUESTED** [review]

File: packages/react/src/components/baseCheckout/index.tsx (hunk 1 above)
Comment (TigranHelio) [reply packages/react/src/components/baseCheckout/index.tsx:244]: isBalanceEnough hints that the variable is a value, not a function. get shows that it is a function which returns isBalanceEnough

File: packages/react/src/components/baseCheckout/index.tsx (hunk 2 above)
Comment (TigranHelio) [reply packages/react/src/components/baseCheckout/index.tsx:113]: same as above

File: packages/react/src/domain/services/AvailableBalanceService.ts
Comment (TigranHelio) [reply packages/react/src/domain/services/AvailableBalanceService.ts:1]: AvailableBalanceService uses values provided by HelioContext

Comment (SurenAvagyanG) [reply packages/react/src/domain/services/AvailableBalanceService.ts:1]: then change it and remove Depdendency from helioContext

File: packages/react/src/ui-kits/tooltip/index.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+interface Props {
+  children: ReactNode;
+  tooltipText: string | ReactNode;
+  placement?: Placement;
+}
+
+export const Tooltip: FC<Props> = ({
+  children,
+  tooltipText,
+  placement = 'top',
```
Comment (00MB) [inline packages/react/src/ui-kits/tooltip/index.tsx:15]: We could probs use this in dashboard now

**Review by 00MB — APPROVED** [review]

File: packages/e2e/src/App.tsx [hunk 4]
```diff
@@ -39,7 +39,7 @@ const App = () => {
         onStartPayment={function (): void {
           console.log('onStartPayment');
         }}
-        supportedCurrencies={['USDC', 'SOL']}
+        // supportedCurrencies={['USDC', 'SOL']}
```
[the remark this answers is not in the document]
Comment (TigranHelio) [reply packages/e2e/src/App.tsx:42]: it is just for testing, we can write whatever we want there

File: packages/react/src/components/heliopayContainer/index.tsx [hunk 5]
```diff
@@ -101,9 +107,26 @@ const HelioPayContainer: FC<HeliopayContainerProps> = ({
...
+          paymentDetails?.normalizedPrice
+        ),
+        tokenSwapQuote,
+      }),
+    [paymentDetails, availableBalances, tokenSwapQuote]
+  );
+
+  const notEnoughFunds =
+    !(isCustomerDetailsRequired || supportedCurrencies?.length) &&
+    !isBalanceEnough;
```
[the remark this answers is not in the document]
Comment (TigranHelio) [reply packages/react/src/components/heliopayContainer/index.tsx:127]: if customer details are required, a modal will open when paying; we will have the available balance check there

**Review by SurenAvagyanG — APPROVED** [review]

**Review by 00e6 — APPROVED** [review]

**Pull request merged**